src/components/Details/TranscriptItem/TranscriptItem.tsx

```diff
@@ -93,9 +100,18 @@ const TranscriptItem: FC<TranscriptItemProps> = ({
   searchQuery,
   handleVideoClick,
   handleTranscriptClick,
+  componentRef,
 }: TranscriptItemProps) => {
+  const transcriptItemRef = useRef<HTMLDivElement>(null);
+  useImperativeHandle(componentRef, () => ({
+    /**Implements componentRef.scrollIntoView by using the inner ref transcriptItemRef */
+    scrollIntoView: () => {
+      transcriptItemRef.current?.scrollIntoView({ behavior: "smooth", block: "center" });
+    },
```

Comment:
Is there a duration to this scrollIntoView? Like the scroll should always take ~1 second or something?

Reply:
There isn't a duration parameter. The duration will depend on the current position and the position of the transcript item. Not sure what to do to make the duration constant regardless of distance. If we make the duration constant, it could lead to a very fast scroll if distance is large. Right now, if the distance is large, the duration is probably long.